### DISCUSSION TITLE
\margintoc vs. \margintocshort

### DISCUSSION THREAD

**Discussion by AlexanderZeilmann** [fmarotta/kaobook#130] [Q&A] [answered]
Is there any reason to use `\margintoc` instead of `\margintocshort` in `\setchapterpreamble[u]`?

From what I can tell, for `\margintoc`, the following two lines do exactly the same:
```
\section[alternative-title-for-toc]{title-as-written-in-text}
\section[alternative-title-for-toc]{title-as-written-in-text}[alternative-title-for-margintoc]
```
whereas `\margintocshort` uses the *alternative-title-for-margintoc* in the margintoc.

So if `\margintocshort` is indeed more general, maybe `\margintoc` should not be mentioned at all in the documentation.

**Comment by fmarotta** [comment, marked as answer]
I totally agree.

A little time ago I made some experiments trying to automatically adjust the width of the box for the section number and improve the justification of the margintoc; since these experiments seem to have worked, I would keep these features and port them to the short margintoc. If you don't mind, I will merge everything into a single command and call it `\margintoc`, just because it is shorter to type and backward-compatible (old users wouldn't need to change all their existing documents adding the 'short' to all their `\margintoc`s). The other side of the coin is that users who, like you, do use `\margintocshort`, would have to change their documents, so maybe I will keep `\margintocshort` as an alias for some time before deprecating it. (Just because I think that the users of `\margintocshort` are a minority, since this command is newer.)

**Reply by AlexanderZeilmann** [reply]
Sounds good to me 😃

**Reply by fmarotta** [reply]
Done
